src/pages/index.js

```diff
@@ -47,14 +73,161 @@ const cardListEl = new Section(
   ".cards__list"
 );
 
-cardListEl.renderItems();
+// old code to delete before submission
+// cardListEl.renderItems();
+
+// Initial Cards
+api
+  .getInitialCards()
+  .then((cards) => {
+    // process the result
+    console.log("Fetched initial cards:", cards);
+    cardListEl.renderItems(cards);
+  })
+  .catch((error) => {
+    // log the error to the console
+    console.error("Error fetching cards", error);
+  });
 
 // Creates an instance of UserInfo class
 const userInfo = new UserInfo({
   nameSelector: ".profile__title",
   descriptionSelector: ".profile__description",
+  avatarSelector: ".profile__image",
 });
 
+// Ensure that user's profile avatar, name, and description persist correctly on page refresh
+// api.getUserInfo() call should be used to set the user info in your UI elements (e.g., text fields or image src attributes)
+document.addEventListener("DOMContentLoaded", () => {
+  api
+    .getUserInfo()
+    .then((userData) => {
+      console.log("User Data:", userData);
+      // Update UI with user data
+      document.querySelector(".profile__image").src = userData.avatar;
+      document.querySelector(".profile__title").textContent = userData.name;
+      document.querySelector(".profile__description").textContent =
+        userData.about;
+    })
+    .catch((error) => {
+      console.error("Error fetching user data", error);
+    });
+});
+
+// UNIVERSAL FORM FUNCTION
+// universal form function that accepts a request, popup instance and optional loading text
+function handleSubmit(request, popupInstance, loadingText = "Saving...") {
+  // here we change the button text
+  popupInstance.renderLoading(true, loadingText);
+  request()
+    .then(() => {
+      // We need to close only in 'then'
+      popupInstance.close();
+    })
+    // we need to catch possible errors
+    .catch(console.error)
+    // in 'finally' return initial button text back in any case
+    .finally(() => {
+      popupInstance.renderLoading(false);
+    });
+}
+
+// Like and unlike
+function handleLikeIcon(card) {
+  if (!card._isLiked) {
```

Comment:
`_isLiked` 

Please, delete the **low dash** (`underscore`) from methods/fields if you want to use them **outside** a class because the `low dash` shows a **private** field which is used only inside the class.

Reply:
on line 142 and 151 does id of the card need to be changed to card.id, not card._id?

Reply:
`_id` may stay with `_` because it comes from the server. But `id` will be much better there

Reply:
> `_isLiked`
> 
> Please, delete the **low dash** (`underscore`) from methods/fields if you want to use them **outside** a class because the `low dash` shows a **private** field which is used only inside the class.

Hi Gennady, I had to change !card.isLiked back to !card._isLiked to fix the like button. I think this created an issue because card has the _isLiked property. Do I still need to find a way to remove the low dash here?